core/domain/user_services_test.py

```diff
@@ -129,3 +133,239 @@ def test_get_user_id_from_username(self):
         # Return None for usernames which don't exist.
         self.assertIsNone(
             user_services.get_user_id_from_username('fakeUsername'))
+
+
+class UpdateContributionMsecTests(test_utils.GenericTestBase):
+    """Test whether contribution date changes with publication of
+    exploration/collection and update of already published exploration/collection
+    """
+
+    EXP_ID = 'test_exp'
+    COL_ID = 'test_col'
+    COLLECTION_TITLE = 'title'
+    COLLECTION_CATEGORY = 'category'
+    COLLECTION_OBJECTIVE = 'objective'
+
+    def test_contribution_msec_updates_on_published_explorations(self):
```

Comment:
Suppose A creates the exploration (and is an owner). A invites B to make commits by assigning them as an editor. B never makes any commits. A publishes the exploration. The test checks that B's first contribution date is not updated.